src/promesa/exec.cljc

```diff
@@ -111,19 +114,20 @@
   ^{:doc "A global, thread per task executor service."
     :no-doc true}
   default-thread-executor
-  #?(:clj  (if virtual-threads-available?
-             (delay (eval '(java.util.concurrent.Executors/newThreadPerTaskExecutor
-                            ^ThreadFactory (promesa.exec/thread-factory))))
-             default-cached-executor)
+  #?(:clj (compile-if-virtual
+           (delay (java.util.concurrent.Executors/newThreadPerTaskExecutor
+                   ^ThreadFactory (promesa.exec/thread-factory)))
+           default-cached-executor)
      :cljs default-executor))
 
 (defonce
   ^{:doc "A global, virtual thread per task executor service."
     :no-doc true}
   default-vthread-executor
-  #?(:clj  (if virtual-threads-available?
-             (delay (eval '(java.util.concurrent.Executors/newVirtualThreadPerTaskExecutor)))
-             default-cached-executor)
+  #?(:clj  (compile-if-virtual
+            (delay (java.util.concurrent.Executors/newThreadPerTaskExecutor
```

Comment:
This should be `java.util.concurrent.Executors/newVirtualThreadPerTaskExecutor` instead of `java.util.concurrent.Executors/newThreadPerTaskExecutor`

Reply:
I think you are probably confused with this part, newThreadPerTaskExecutor constructs platform threads and the `newVirtualThreadPerTaskExecutor` constructs virtual threads

Reply:
Pushed.